File: flowquery-py/src/parsing/operations/delete_relationship.py
"""Represents a DELETE operation for deleting virtual relationships."""

from typing import Any, Dict, List

from ...graph.database import Database
from ...graph.relationship import Relationship
from .operation import Operation


class DeleteRelationship(Operation):
    """Represents a DELETE operation for deleting virtual relationships."""

    def __init__(self, relationship: Relationship) -> None:
        super().__init__()
        self._relationship = relationship

    @property
    def relationship(self) -> Relationship:
        return self._relationship

    async def run(self) -> None:
        if self._relationship is None:
            raise ValueError("Relationship is null")
        db = Database.get_instance()
        db.remove_relationship(self._relationship)

    @property
    def results(self) -> List[Dict[str, Any]]:
        return []

File: src/parsing/operations/delete_relationship.ts
import Database from "../../graph/database";
import Relationship from "../../graph/relationship";
import Operation from "./operation";

class DeleteRelationship extends Operation {
    private _relationship: Relationship | null = null;
    constructor(relationship: Relationship) {
        super();
        this._relationship = relationship;
    }
    public get relationship(): Relationship | null {
        return this._relationship;
    }
    public run(): Promise<void> {
        return new Promise(async (resolve, reject) => {
            try {
                if (this._relationship === null) {
                    throw new Error("Relationship is null");
                }
                const db = Database.getInstance();
                db.removeRelationship(this._relationship);
                resolve();
            } catch (error) {
                reject(error);
            }
        });
    }
    public get results(): Record<string, any>[] {
        return [];
    }
}
export default DeleteRelationship;
